build-content-management-system.mdx

```diff
@@ -0,0 +1,451 @@
+---
+title: 'Create your own content management system with with Remix and Xata'
+description: 'Learn how to create a custom CMS using Xata, Remix, Novel, LiteLLM, and Vercel.'
+image:
+  src: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/gallery-cms-astro-cloudflare.png
+  alt: Xata CMS
+author: Rishi Raj Jain
+date: 02-01-2024
+published: true
+slug: content-management-system-remix-xata
+ogImage: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/gallery-cms-astro-cloudflare.png
+---
+
+In this post, you'll create a content CMS using Xata, Remix, Novel, LiteLLM, and Vercel. You'll learn how to:
+
+- Set up Xata
+- Create a schema with different column types
+- Handle forms in Remix using Form Actions
+- Implement Client Side Image Uploads
+- Use an AI-powered WYSIWYG Editor
+- Implement content-wide search
+- Create dynamic content routes with Remix
+
+## Before you begin
+
+### Prerequisites
+
+You'll need the following:
+
+- A [Xata](https://xata.io/) account
+- [Node.js 18](https://nodejs.org/en/blog/announcements/v18-release-announce) or later
+- An [OpenAI](https://platform.openai.com) account
+- A [Vercel](https://vercel.com) Account
+
+### Tech Stack
+
+| Technology                         | Description                                                                     |
+| ---------------------------------- | ------------------------------------------------------------------------------- |
+| https://xata.io                    | Serverless database platform for scalable, real-time applications.              |
+| https://remix.run                  | Framework for building full-stack web applications with focus on Web Standards. |
+| https://github.com/BerriAI/litellm | Call all LLM APIs using the OpenAI format.                                      |
+| https://novel.sh                   | A Notion-style WYSIWYG editor with AI-powered autocompletion                    |
+| https://tailwindcss.com/           | CSS framework for building custom designs                                       |
+| https://vercel.com                 | A cloud platform for deploying and scaling web applications.                    |
```

Comment:
Not a big deal but these could be links instead of direct URLs.

```suggestion
| Technology                         | Description                                                                     |
| ---------------------------------- | ------------------------------------------------------------------------------- |
| [Xata](https://xata.io)                    | Serverless database platform for scalable, real-time applications.              |
| [Remix](https://remix.run)                  | Framework for building full-stack web applications with focus on Web Standards. |
| [litelln](https://github.com/BerriAI/litellm) | Call all LLM APIs using the OpenAI format.                                      |
| [Nove](https://novel.sh)                   | A Notion-style WYSIWYG editor with AI-powered autocompletion                    |
| [TailwindCSS](https://tailwindcss.com/)           | CSS framework for building custom designs                                       |
| [Vercel](https://vercel.com)                 | A cloud platform for deploying and scaling web applications.                    |
```

Reply:
Exactly what @tsg said.